ToDo.UI/FormsApp.xaml.cs

```diff
@@ -14,6 +15,131 @@ public partial class FormsApp : Application
         public FormsApp()
         {
             InitializeComponent();
+
+            Resources.Add
```

Comment:
Is there any specific reason why you create styles in code-behind and not in the xaml?

We do very much prefer the XAML approach

Reply:
First I added them into the XAML but then I wanted to set some values depending on the platform.
(eg: Value = Device.RuntimePlatform == Device.iOS ? Color.Black : Color.WhiteSmoke)
And then the idea was to keep the styles together so I moved all of them to the code-behind.

Should I move the basic ones back to XAML? Or all of them? I don't know how to set a value in XAML depending on the platform.

Reply:
Haaave you met `OnPlatform`? It is your friend 😄 

https://www.xamarinhelp.com/xamarin-forms-onplatform-runtimeplatform/

Reply:
Yes please do so.
You can set properties in XAML based on the platform using `<OnPlatform>`
https://docs.microsoft.com/en-us/xamarin/xamarin-forms/xaml/xaml-basics/essential-xaml-syntax#platform-differences-with-onplatform

You can set properties in XAML based on idiom with `<OnIdiom>` (Desktop, Tablet, Phone)

You can even use them *combined* 😎
https://stackoverflow.com/questions/40111252/is-it-possible-to-mix-onidiom-and-onplatform-in-xaml-xamarin-forms

```xaml
<ColumnDefinition>
  <ColumnDefinition.Width>
    <OnIdiom x:TypeArguments="GridLength">
      <OnIdiom.Phone>
        <OnPlatform x:TypeArguments="GridLength" iOS="*" Android="*" WinPhone="100" />
      </OnIdiom.Phone>
      <OnIdiom.Tablet>
        <OnPlatform x:TypeArguments="GridLength" iOS="*" Android="*" WinPhone="200" />
      </OnIdiom.Tablet>
    </OnIdiom>
  </ColumnDefinition.Width>
</ColumnDefinition>

```

Reply:
Okay we became friends with OnPlatform now :)
8ef7800cb659827fc65304a9520aec9e7962a845